### PR TITLE
Minor cleanups

### PR DESCRIPTION
**Pull request by Goooler**
## Description

Follow up #3089.

## Type of change

:white_check_mark: General change (non-breaking change that doesn't fit the below categories like copyediting)
:x: Bug fix (non-breaking change which fixes an issue)
:x: New feature (non-breaking change which adds functionality)
:x: Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: lawnchair/src/app/lawnchair/font/FontCache.kt [hunk 1]
```diff
@@ -159,9 +159,10 @@ class FontCache private constructor(private val context: Context) {
             fun fromJsonString(context: Context, jsonString: String): Font {
                 val obj = JSONObject(jsonString)
                 val className = obj.getString(KEY_CLASS_NAME)
-                val clazz = Class.forName(className)
-                val constructor = clazz.getMethod("fromJson", Context::class.java, JSONObject::class.java)
-                return constructor.invoke(null, context, obj) as Font
+                return Class.forName(className)
+                    .getMethod("fromJson", Context::class.java, JSONObject::class.java)
+                    .apply { isAccessible = true }
```
Comment (Goooler) [inline lawnchair/src/app/lawnchair/font/FontCache.kt:165]: Set `isAccessible` to true for all reflection usages.

File: lawnchair/src/app/lawnchair/preferences/PrefLifecycleObserver.kt [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
-import androidx.lifecycle.OnLifecycleEvent
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
 
 class PrefLifecycleObserver<T>(
     private val prefEntry: PrefEntry<T>,
     private val onChange: Runnable
-) : LifecycleObserver, PreferenceChangeListener {
-
-    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
```
Comment (Goooler) [inline lawnchair/src/app/lawnchair/preferences/PrefLifecycleObserver.kt:12]: `OnLifecycleEvent` is deprecated, migrate to `DefaultLifecycleObserver`

File: lawnchair/src/app/lawnchair/ui/TouchTarget.kt [hunk 3]
```diff
@@ -16,6 +16,7 @@
 
 package app.lawnchair.ui
 
+import android.widget.Button
```
Comment (yasanglass) [inline lawnchair/src/app/lawnchair/ui/TouchTarget.kt]: This is the _View_ Button class, it looks like the doc is referencing a _composable_ Button.

Comment (Goooler) [reply lawnchair/src/app/lawnchair/ui/TouchTarget.kt]: 949d54d2781bb47f2a8b1fcc9553d3ba43178933

File: lawnchair/src/app/lawnchair/util/ViewUtils.kt [hunk 4]
```diff
@@ -1,20 +1,6 @@
...
-fun ViewGroup.getAllChildren(list: MutableList<View>) {
-    for (i in (0 until childCount)) {
-        val child = getChildAt(i)
-        if (child is ViewGroup) {
-            child.getAllChildren(list)
-        } else {
-            list.add(child)
-        }
-    }
-}
```
Comment (Goooler) [inline lawnchair/src/app/lawnchair/util/ViewUtils.kt:17]: Dup with `recursiveChildren`

https://github.com/LawnchairLauncher/lawnchair/blob/8263c6feb0aebb0fc607903a6295aa98863f2d6d/lawnchair/src/app/lawnchair/util/LawnchairUtils.kt#L143-L147

File: lawnchair/src/app/lawnchair/util/OptionalUtil.kt [hunk 5]
```diff
@@ -1,9 +0,0 @@
-package app.lawnchair.util
-
-import java.util.*
-
-val <T> Optional<T>.value: T?
-    get() {
-        if (!isPresent) return null
-        return get()
-    }
```
Comment (Goooler) [inline lawnchair/src/app/lawnchair/util/OptionalUtil.kt:9]: https://kotlinlang.org/docs/whatsnew17.html#new-experimental-extension-functions-for-java-optionals

**Pull request merged**